### PR TITLE
added authDAO, loginDAO, postDAO, and key

### PR DESCRIPTION
**Pull request by camrcast**
added authDAO, loginDAO, postDAO, and key

### CODE REVIEW

File: src/key.txt
Comment (hubmelco) [inline src/key.txt:1]: I wouldn't push keys that encrypt stuff to a public space

File: src/repository/authDAO.js
Comment (hubmelco) [inline src/repository/authDAO.js:1]: This would be middleware, not a DAO. It doesn't really do anything in the database. You can also tell that it's middleware because of the use of req, res, and next as function arguments

File: src/repository/loginDAO.js [hunk 1]
```diff
@@ -24,4 +34,25 @@ async function createToken(user){
         }
     );
     return token;
+}
+
+async function findUser(username){
```
Comment (hubmelco) [inline src/repository/loginDAO.js:39]: This would be part of a userDAO

File: src/repository/loginDAO.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
 
+let key = "";
+fs.readFile('./src/key.txt', (err, data) => {
+    if (err) throw err;
+    key = data.toString();
+});
+
+const tableName = "";
+
 async function login(username, password){
```
Comment (hubmelco) [inline src/repository/loginDAO.js:13]: This would be part of a service. Could be a separate login service or it could be included in a user service, but similarly to the middleware, it's main goal is getting a token from JWT and not database interaction so I wouldn't make it a DAO

File: src/repository/postDAO.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const uuid = require("uuid");
+const { PutCommand } = require("@aws-sdk/lib-dynamodb");
+
+const tableName = "";
+
+async function createPost(username, text, score){
```
Comment (hubmelco) [inline src/repository/postDAO.js:6]: This would be a service file function because it is handling user data. I would separate it from the DAO

File: src/repository/postDAO.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    if (score < 0 || !text){
+        return 1;
```
Comment (hubmelco) [inline src/repository/postDAO.js:8]: returning numbers can be vague and hard to follow if you don't know what the developer who wrote it was thinking. You can maybe throw an error here or return an object such as {message: "Invalid score or text inputted"};

**Review by hubmelco — COMMENTED** [review]
Overall, nice code. You don't have to fix the auth and login comments I made since I said I'd do it as part of my admin task, but before we merge the PR I'd like some test to be made at least for the creating a post section

File: src/repository/authDAO.js
Comment (camrcast) [reply src/repository/authDAO.js:1]: I will remove the DAO part of the naming

File: src/repository/postDAO.js (hunk 4 above)
Comment (camrcast) [reply src/repository/postDAO.js:8]: Good point. I will look into return objects with messages.
